### PR TITLE
Deprecate TSSslSecretGet (not thread safe).

### PR DESCRIPTION
**Pull request by ywkaras**
The current version is not thread safe because it returns a pointer into a mutex-protected data structure, which may be invalidated by another thread

### CODE REVIEW

**Comment by bneradt** [comment]
Looks good. It's good to give a heads-up to people that this function signature will be changing in 10.x.

I suggest providing a brief description of the new signature now in these comments.

**Comment by ywkaras** [comment]
The description is available in the Rel 10 documentation though.

**Comment by maskit** [comment]
Do we deprecate things on minor releases? I feel like we've been doing it only on major releases but I'm not sure.
If it's deprecated on 9.2.0, what's the alternative way that is not deprecated on the version?

**Comment by maskit** [comment]
Is the change #8790? It isn't even approved. Do we have consensus on the way of depreciation, which only changes the return type and the arguments?

I also found that `TSSslSecretGet` was introduced by #6609 and its milestone is 9.2.0, which means the API haven't been released publicly. If it's true, I think we can just remove the API.

File: include/ts/ts.h
Comment (maskit) [inline include/ts/ts.h:1307]: typo: is is -> it is

**Review by maskit — CHANGES_REQUESTED** [review]
I'm not sure if this deprecation is needed in the first place.

The typo should be fixed anyway.

**Comment by ywkaras** [comment]
> Is the change #8790? It isn't even approved. Do we have consensus on the way of depreciation, which only changes the return type and the arguments?
> 
> I also found that `TSSslSecretGet` was introduced by #6609 and its milestone is 9.2.0, which means the API haven't been released publicly. If it's true, I think we can just remove the API.

I was told not to mark #8790 for back port to 9.2.  If #8790 is rejected, we will have to eliminate or replace TSSslSecretGet() in another PR, it is not thread safe.  #8790 is currently running in Yahoo prod, I was waiting for it to soak for a while before trying to merge it in open source.

**Comment by maskit** [comment]
> I was told not to mark https://github.com/apache/trafficserver/pull/8790 for back port to 9.2

I wonder why. Backporting it to 9.2 is another option indeed.

**Comment by ywkaras** [comment]
> > I was told not to mark #8790 for back port to 9.2
> 
> I wonder why. Backporting it to 9.2 is another option indeed.

@bryancall do we want to reconsider back porting?

File: include/ts/ts.h
Comment (ywkaras) [reply include/ts/ts.h:1308]: TS_MAYBE

**Comment by ywkaras** [comment]
This won't be needed if https://github.com/apache/trafficserver/pull/8854 makes it into 9.2.

**Pull request closed without merging**
